rasa/core/domain.py

```diff
@@ -344,6 +373,11 @@ def collect_templates(
         templates = {}
         for template_key, template_variations in yml_templates.items():
             validated_variations = []
+            if template_variations is None:
+                raise InvalidDomain(
+                    "Utterance '{}' does not have any defined templates.".format(template_key)
```

Comment:
@MetcalfeTom I changed this one since this is what's really being checked here, whether or not it has `- text` (or `- custom`), is checked below if the template exists